service/worker/scanner/scanner_test.go

```diff
@@ -40,9 +64,249 @@ func TestScannerSuite(t *testing.T) {
+			cfg: Config{
+				Persistence: &config.Persistence{
+					DefaultStore: "nosql",
+					DataStores: map[string]config.DataStore{
+						"nosql": {
+							NoSQL: &config.NoSQL{},
+						},
+					},
+				},
+				TaskListScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+					return false
+				},
+				HistoryScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+					return true
+				},
+			},
+			setupMocks: func() {
+				s.mockWorker.EXPECT().Start().Return(nil).Times(1)
+			},
+		},
+		{
+			name: "failed to start worker",
+			cfg: Config{
+				Persistence: &config.Persistence{
+					DefaultStore: "nosql",
+					DataStores: map[string]config.DataStore{
+						"nosql": {
+							NoSQL: &config.NoSQL{},
+						},
+					},
+				},
+				TaskListScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+					return false
+				},
+				HistoryScannerEnabled: func(opts ...dynamicconfig.FilterOption) bool {
+					return true
+				},
+			},
+			setupMocks: func() {
+				s.mockWorker.EXPECT().Start().Return(errors.New("some new error")).Times(1)
+			},
+			err: errors.New("some new error"),
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Run(tc.name, func(t *testing.T) {
+			defer goleak.VerifyNone(s.T())
+
+			s.scanner.context.cfg = tc.cfg
+			tc.setupMocks()
+
+			err := s.scanner.Start()
+
+			if tc.err != nil {
+				s.Error(err)
+				s.Equal(tc.err, err)
+			} else {
+				s.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *scannerTestSuite) Test__startWorkflow() {
+	testCases := []struct {
+		name         string
+		options      client.StartWorkflowOptions
+		workflowType string
+		args         interface{}
+		setupMocks   func(client.StartWorkflowOptions, string, interface{})
+		err          error
+	}{
+		{
+			name: "success with args",
+			options: client.StartWorkflowOptions{
+				ID: "testIDWithArgs",
+			},
+			workflowType: "workflowTypeWithArgs",
+			args:         "testArgs",
+			setupMocks: func(options client.StartWorkflowOptions, workflowType string, args interface{}) {
+				s.mockClient.On("StartWorkflow", mock.Anything, options, workflowType, args).Return(nil, nil)
+			},
+		},
+		{
+			name: "success with no args",
+			options: client.StartWorkflowOptions{
+				ID: "testIDWithNoArgs",
+			},
+			workflowType: "workflowTypeWithArgs",
+			setupMocks: func(options client.StartWorkflowOptions, workflowType string, args interface{}) {
+				s.mockClient.On("StartWorkflow", mock.Anything, options, workflowType).Return(nil, nil)
+			},
+		},
+		{
+			name: "WorkflowExecutionAlreadyStarted error",
+			options: client.StartWorkflowOptions{
+				ID: "testIDWorkflowExecutionAlreadyStarted",
+			},
+			workflowType: "workflowTypeWorkflowExecutionAlreadyStarted",
+			setupMocks: func(options client.StartWorkflowOptions, workflowType string, args interface{}) {
+				s.mockClient.On("StartWorkflow", mock.Anything, options, workflowType).Return(nil, &shared.WorkflowExecutionAlreadyStartedError{})
+			},
+		},
```

Comment:
personally: since this is a bit of an exception (return err but expect no err), I like having an explicit
```go
err: nil,
```
to show it was thought about and is nil on purpose.